ACE/ace/Condition_Recursive_Thread_Mutex.h

```diff
@@ -101,11 +108,25 @@ class ACE_Export ACE_Condition<ACE_Recursive_Thread_Mutex>
   ACE_Recursive_Thread_Mutex &mutex_;
 
 };
+// prevent implicit instantiations by includers to relieve the linker
+#if defined (ACE_HAS_CPP11_EXTERN_TEMPLATES)
+// suppress a warning, g++ 5.2.1 does not support attributes on template
+// instantiation declarations. *TODO*: this may go back further
+# if defined (__GNUG__) && (__GNUC__ == 5 && __GNUC_MINOR__ == 2 && __GNUC_PATCHLEVEL__ == 1)
```

Comment:
Also use a new define for this, that makes it easier when this is needed in more places and for example gcc 5.2.2 has the same issue